### PR TITLE
Feature/tdd project

### PR DESCRIPTION
**Pull request by Diegogagan2587**
# Implented Changes
Include unit test and solution for the next algorithms:
- fizz buzz
- reverse a string
- factorial number

It was followed the TDD methodolgy.

### CODE REVIEW

File: test/fizzbuzz_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  it 'Should return an string when input is 1' do
+    fizzbuzz = Fizzbuzz.new(1).fizz_buzz
+    expect(fizzbuzz).to be_a(String)
+  end
+
+  it 'Should return 1, 2, Fizz, 4, Buzz when input is 5' do
+    fizzbuzz = Fizzbuzz.new(5).fizz_buzz
+    expect(fizzbuzz).to eql('1, 2, Fizz, 4, Buzz')
+  end
+end
```
Comment (binhussen) [inline test/fizzbuzz_spec.rb]: - Kindly, would you test `fizzbuzz` when `N`  divisible is `3` `5` both `3 & 5` such as `15` and other number to fit the requirement 👍

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @Diegogagan2587 and @sergio-usma , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Fo,lowed TDD rules ✔️
- Well Structured Code ✔️
- Used Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/fizzbuzz_spec.rb (hunk 1 above)
Comment (Diegogagan2587) [reply test/fizzbuzz_spec.rb]: @binhussen Thanks for pointing that out, we'll make the corrections!

**Review by danmainah — APPROVED** [review]
Hi @Diegogagan2587 and @sergio-usma , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to add more test cases for the fizzbuzz method.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**